cmake/BLTOptions.cmake

```diff
@@ -93,6 +99,10 @@ if( NOT ( ( BLT_CXX_STD STREQUAL "c++98" )
 Valid Options are ( c++98, c++11, c++14 )")
 endif()
 
+set(FORTRAN_FORMAT FREE CACHE STRING
+  "Fortran source formatting: FREE (default), FIXED, or INFER")
+set_property(CACHE FORTRAN_FORMAT PROPERTY STRINGS FREE FIXED INFER)
```

Comment:
How often do we expect users to want to explicitly set this option?
Should this be marked as an advanced cache variable?

Reply:
I was surprised and confused that you were setting FREE for everything, and others may have a similar experience. Marking as advanced may delay the resolution of peoples issues... just food for thought. Personally I think INFER should be the default, but it's your project, so I'll defer to your preferences.

Reply:
After an internal (offline) discussion, we're leaning towards making ``INFER`` the default, and projects that want to change this default can explicitly set the desired behavior.

This seems likely to lead to fewer surprises for fortran projects.

If the default is ``INFER``, would it make sense for ``FORTRAN_FORMAT`` to be marked as advanced?

Reply:
I think it could go either way.

I personally tend to reserve advanced CMake options for ones targeting the
project’s developers most of the time, or ones that are blatantly
experimental or dangerous (like in source builds). Basically anything where
a particular user requested some non standard functionality etc.

In this vein I’m inclined to default any options that a project provides to
non-advanced. In my mind, advanced says “you’re on your own if you mess with
this and we cannot envision a justifiable reason for doing so.” In
practice, however, I find most of the time users don’t even know about
`ccmake` or `cmake-gui` and therefore advanced vs normal is of very little
impact.

Reply:
After reviewing -- I still feel we should simply remove all of the logic related to this and let folks using 
BLT set `CMAKE_Fortran_FORMAT`  (or the related source properties) as they see fit in their project. 

Adding the option is another thing to explain and support vs standard CMake, which in this case is  very straight forward. 

Again, sorry for the extra effort, but not having it means we save on docs, maintenance, testing, etc in the long run. 

@kennyweiss @white238, what do you think?

Reply:
The benefit of keeping it is that the code is already written and it would make it easy for people setting up a fortran code to know about this distinction.  

The downside is having to describe/maintain it, and the added complexity of additional configuration options.

I'm ok with removing this feature.